sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -421,6 +422,119 @@ ng_ipv6_addr_t *ng_ipv6_netif_find_best_src_addr(kernel_pid_t pid, const ng_ipv6
     return _match_prefix(pid, dest, true);
 }
 
+/* TODO: put this somewhere more central and L2 protocol dependent */
+#define IID_LEN (8)
+
+static bool _hwaddr_to_iid(uint8_t *iid, const uint8_t *hwaddr, size_t hwaddr_len)
```

Comment:
I actually would like to see this publicly available for other layers. The rationale is that I would use such a function in RPL, where I need to add an ipv6 address to the interface (prefix given during runtime). Maybe with some more abstractions so that I do not have to deal with hardware addresses in RPL (: